From a5e5baf1c92df4bb16aee42a6e160b44abef75bf Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 15 Jan 2024 15:21:05 +0100
Subject: [PATCH 0447/1695] dts: stm32mp13: Define scmi regulator consumers

Describe the voltage regulators exposed as SCMI voltage domains to
non-secure world using a st,scmi-regulator-consumer compatible node.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 19ce038c97a7e0d25510aeec175dbbb02b0067a6 dts: stm32mp13: Define scmi regulator consumers]
Change-Id: I396a201e978965da4d6a9dcaf989de2a08d3f488
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi    | 23 ++++++++++++++++++
 core/arch/arm/dts/stm32mp135f-dk.dts | 35 ++++++++++++++++++++++++++++
 2 files changed, 58 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index 2b6bcbdc6..6800c5f17 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -83,6 +83,29 @@
 
 	};
 
+	scmi_regu: scmi-regu {
+		compatible = "st,scmi-regulator-consumer";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		scmi-channel-id = <0>;
+
+		scmi_reg11: voltd-reg11 {
+			reg = <VOLTD_SCMI_REG11>;
+			voltd-supply = <&reg11>;
+		};
+
+		scmi_reg18: voltd-reg18 {
+			reg = <VOLTD_SCMI_REG18>;
+			voltd-supply = <&reg18>;
+		};
+
+		scmi_usb33: voltd-usb33 {
+			reg = <VOLTD_SCMI_USB33>;
+			voltd-supply = <&usb33>;
+		};
+	};
+
 	sdmmc1_io: sdmmc1_io {
 		compatible = "st,stm32mp13-iod";
 		regulator-name = "sdmmc1_io";
diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index 18a60fde0..684f5b314 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -510,6 +510,41 @@
 	status = "okay";
 };
 
+&scmi_regu {
+	voltd-vddcpu {
+		reg = <VOLTD_SCMI_STPMIC1_BUCK1>;
+		voltd-supply = <&vddcpu>;
+	};
+	voltd-vdd {
+		reg = <VOLTD_SCMI_STPMIC1_BUCK3>;
+		voltd-supply = <&vdd>;
+	};
+	voltd-vddcore {
+		reg = <VOLTD_SCMI_STPMIC1_BUCK4>;
+		voltd-supply = <&vddcore>;
+	};
+	voltd-vdd-adc {
+		reg = <VOLTD_SCMI_STPMIC1_LDO1>;
+		voltd-supply = <&vdd_adc>;
+	};
+	voltd-vdd-usb {
+		reg = <VOLTD_SCMI_STPMIC1_LDO4>;
+		voltd-supply = <&vdd_usb>;
+	};
+	voltd-vdd-sd {
+		reg = <VOLTD_SCMI_STPMIC1_LDO5>;
+		voltd-supply = <&vdd_sd>;
+	};
+	voltd-v1v8-periph {
+		reg = <VOLTD_SCMI_STPMIC1_LDO6>;
+		voltd-supply = <&v1v8_periph>;
+	};
+	voltd-v3v3-sw {
+		reg = <VOLTD_SCMI_STPMIC1_PWR_SW2>;
+		voltd-supply = <&v3v3_sw>;
+	};
+};
+
 &sdmmc1_io {
 	vddsd1-supply = <&vdd>;
 };
-- 
2.47.2

